### PR TITLE
Disable chef-client service if it exists

### PR DESCRIPTION
**Pull request by shoekstra**
### Description

If using a scheduled task then the service should be disabled if it exists.

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by shoekstra** [comment]
This needs some more testing, please don't merge just yet.

**Comment by shoekstra** [comment]
Good to merge now, and fixed the spec tests.

File: recipes/cron.rb [hunk 1]
```diff
@@ -87,7 +87,7 @@ class ::Chef::Recipe
 
 # Generate a uniformly distributed unique number to sleep.
 if node['chef_client']['splay'].to_i > 0
-  checksum   = Digest::MD5.hexdigest(node['node_name'])
+  checksum   = Digest::MD5.hexdigest(node['hostname'])
```
Comment (tas50) [inline recipes/cron.rb]: Can you back this part out. I'm going to merge a more robust fix in #467

**Comment by tas50** [comment]
I fixed it via the GUI. Thanks for fixing the service issue though

**Pull request merged**